modules/fedimint-gw-client/src/pay.rs

```diff
@@ -0,0 +1,17 @@
+/// State machine that pays a lightning invoice on behalf of a federation user.
+///
+/// ```mermaid
+/// graph LR
+/// classDef virtual fill:#fff,stroke-dasharray: 5 5
+///
+///    PayInvoice -- validate outgoing contract --> Canceled
+///    PayInvoice -- await preimage offer accepted --> BuyPreimage
+///    PayInvoice -- await preimage offer failure --> Canceled
+///    BuyPreimage -- await federation decryption --> Preimage
```

Comment:
This does a bit more than just waiting, it actually has to create another transaction funding an incoming contract and then wait for this to trigger a preimage decryption.

Reply:
I split this into two states. First transition awaits for the incoming contract funding tx to be accepted, which transitions into the Funded state. Then we await preimage decryption after we're in the Funded state.